application/common/components/Message.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+
+namespace common\components;
+
+
+use yii\mail\BaseMessage;
+use yii\mail\MessageInterface;
+
+class Message extends BaseMessage
+{
+    /** @var string */
+    private $_charset;
+
+    /** @var string */
+    private $_from;
+
+    /** @var string[] */
+    private $_to;
+
+    /** @var string[] */
+    private $_replyTo;
+
+    /** @var string[] */
+    private $_cc;
+
+    /** @var string[] */
+    private $_bcc;
+
+    /** @var string */
+    private $_subject;
+
+    /** @var string */
+    private $_textBody;
+
+    /** @var string */
+    private $_htmlBody;
+
+    /**
+     * @inheritdoc
+     */
+    public function getCharset()
+    {
+        return $this->_charset ?? 'UTF-8';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setCharset($charset)
+    {
+        $this->_charset = $charset;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getFrom()
+    {
+        return $this->_from;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setFrom($from)
+    {
+        if (is_array($from) && count($from) > 0) {
+            $this->_from = $from[0];
+        } else {
+            $this->_from = $from;
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getTo()
+    {
+        return $this->_to;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setTo($to)
+    {
+        if (is_array($to)) {
+            $this->_to = $to;
+        } else {
+            $this->_to = explode(",", $to);
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getReplyTo()
+    {
+        return $this->_replyTo ?? [$this->_from];
```

Comment:
Minor suggestion: It might be good to use `getFrom()` as the fallback (instead of `_from`) in case any fallback logic is ever added to `getFrom()`... that way we'd get the benefit of it here, too.
```suggestion
        return $this->_replyTo ?? [$this->getFrom()];
```

Reply:
good idea